connMgr.js

```diff
@@ -50,10 +58,17 @@ class ConnectionManager {
         offset = this.doHandshake(buff, offset);
       }
       if (this.handshakeState === 'handshake-done') {
-        // while (offset < buff.length) {
-        while (buff.length - offset > 2) {
+        while (offset < buff.length) {
           // print(`\tbuff.length=${buff.length}, offset=${offset}`);
-          [err, offset] = this.processMessage(buff, offset);
+          tryCatch(
+            () => {
+              [err, offset] = this.processMessage(buff, offset);
```

Comment:
Why does this method return a result type (err, offset) and throw an error? Doing a try/catch here doesn't seem like a temporary patch..

Would it be better to improve this method to not throw?

Reply:
Yes indeed, mixing two patterns (returning err and throwing err) looks bad...
I changed the dependencies not to throw exceptions so now `this.processMessage()' does not throw.
https://github.com/media-fish/rtmp-server/pull/11